server/xcode/xcode_darwin.go

```diff
@@ -126,13 +140,51 @@ func (x *xcodeLocator) locate() error {
 				continue
 			}
 			log.Debugf("Mapped Xcode Version %s=>%s with SDKs %+v", version, developerDirPath, sdks)
-			versionMap[version] = &xcodeVersion{
+			xv := &xcodeVersion{
 				version:          mostPreciseVersion,
 				developerDirPath: developerDirPath,
 				sdks:             sdks,
 			}
+			versionMap[version] = xv
+			if defaultDeveloperDir == developerDirPath {
+				versionMap[""] = xv
+			}
 		}
 	}
+
+	if xv, ok := versionMap[""]; ok {
+		log.Infof("xcode-select set to Xcode version %s", xv.version)
+	} else {
+		// If we are here then defaultDeveloperDir is not a valid Xcode path
+		if defaultDeveloperDir == "/Library/Developer/CommandLineTools" {
+			var version string
+			out, err := exec.Command("sh", "-c", "pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '/version:/ {print $2}'").Output()
+			if err == nil {
+				version = strings.TrimSpace(string(out))
+			} else {
+				version = "unknown"
+			}
+			log.Infof("xcode-select set to Command Line Tools (%s)", version)
+
+			sdkPaths, err := fs.Glob(os.DirFS(defaultDeveloperDir), "SDKs/*")
+			if err != nil {
+				log.Warningf("Error reading CommandLineTools SDKs from %s: %s", defaultDeveloperDir, err.Error())
+			} else {
+				sdks := make(map[string]string)
+				for _, sdkPath := range sdkPaths {
+					sdks[strings.TrimSuffix(filepath.Base(sdkPath), ".sdk")] = sdkPath
+				}
+				versionMap[""] = &xcodeVersion{
+					version:          "",
+					developerDirPath: defaultDeveloperDir,
+					sdks:             sdks,
+				}
+			}
```

Comment:
it's [idomatic](https://zchee.github.io/golang-wiki/CodeReviewComments/#indent-error-flow) in go to return early and unnest the else case. If we moved this block to a function, we could do that. Something like

func commandlineToolVersion() *xcodeVersion {
 ...
}